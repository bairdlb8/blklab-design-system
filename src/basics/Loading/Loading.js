import React from 'react'
import PropTypes from 'prop-types'

import './Loading.scss'

const Loading = ({ className = '' }) => {
  return (
    <div className="loading">
      <div className="double">
        <div className="double-bounce1"></div>
        <div className="double-bounce2"></div>
      </div>
    </div>
  )
}

Loading.propTypes = {}

export default Loading
