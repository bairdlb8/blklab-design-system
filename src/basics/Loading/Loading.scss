.loading {
  position: absolute;
  left: 50%;
  top: 50%;
  transform: translateX(-50%) translateY(-50%);

  .double {
    width: 1.5em;
    height: 1.5em;
    position: relative;
  }

  .double-bounce1,
  .double-bounce2 {
    width: 100%;
    height: 100%;
    border-radius: 50%;
    background-color: var(--primary-color);
    opacity: 0.6;
    position: absolute;
    top: 0;
    left: 0;
    animation: sk-bounce 2s infinite ease-in-out;
  }

  .double-bounce2 {
    animation-delay: -1s;
  }

  @keyframes sk-bounce {
    0%,
    100% {
      transform: scale(0);
    }
    50% {
      transform: scale(1);
    }
  }
}
