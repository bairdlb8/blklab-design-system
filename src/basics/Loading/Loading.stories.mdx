import { Meta, Story, Props, Preview } from '@storybook/addon-docs/blocks'

import Loading from './Loading'

<Meta title="Basics / Loading" component={ Loading } />

# Loading

## Add some text here

export const Template = (args) => {
  return <Loading {...args} />
}

<Preview withToolbar>
  <Story name="Primary" args={ {} }>
    {Template.bind({})}
  </Story>
</Preview>

<Props of={ Loading } />
