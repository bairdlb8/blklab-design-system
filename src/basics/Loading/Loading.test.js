import React from 'react'
import { render } from '@testing-library/react'
import '@testing-library/jest-dom/extend-expect'
import { axe, toHaveNoViolations } from 'jest-axe'
import Loading from './Loading'

expect.extend(toHaveNoViolations)

describe('<Loading />', () => {
  describe('accessibility', () => {
    it('should not fail any accessibility tests', async () => {
      const { container } = render(<Loading />)
      expect(await axe(container)).toHaveNoViolations()
    })
  })

  it('should take a snapshot', () => {
    const { asFragment } = render(<Loading />)

    expect(asFragment(<Loading />)).toMatchSnapshot()
  })
})
