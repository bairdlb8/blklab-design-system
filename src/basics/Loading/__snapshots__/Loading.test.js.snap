// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Loading /> should take a snapshot 1`] = `
<DocumentFragment>
  <div>
    Loading
  </div>
</DocumentFragment>
`;
